src/mad_spark_alt/evolution/semantic_operators.py

```
@@ -533,9 +588,13 @@ def _parse_batch_response(self, response: str, expected_count: int) -> List[str]
                     break
             else:
-                # Fallback if pattern not found
-                logger.warning(f"Could not find mutation for IDEA_{i+1}")
-                mutations.append(f"Enhanced version of idea {i+1}")
+                # Fallback if pattern not found - create meaningful variation
+                logger.warning(f"Could not find mutation for IDEA_{i+1}, using fallback")
+                if original_ideas and i < len(original_ideas):
+                    original_content = original_ideas[i].content
+                    mutations.append(f"Enhanced variation of original concept: Building upon '{original_content[:50]}...', this mutation explores alternative implementation strategies while maintaining the core objective. The approach introduces different methodologies, tools, and frameworks to achieve similar outcomes through a varied pathway. By shifting perspectives on scale, audience, or technological approach, this variation demonstrates how the fundamental concept can be realized through innovative alternatives.")
+                else:
+                    mutations.append(f"Enhanced variation exploring alternative approaches: This mutation investigates different implementation strategies while maintaining the core objective of the original idea. The approach introduces alternative methodologies, tools, and frameworks to achieve similar outcomes through a different pathway. By exploring varied perspectives such as changing the scale of implementation, shifting target audience, or adopting different technological foundations, this variation demonstrates how the same fundamental problem can be addressed through multiple viable and innovative solutions.")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)